### DISCUSSION TITLE
Add a playground to the docs

### DISCUSSION THREAD

**Discussion by nandorojo** [0no-co/gql.tada#130] [Ideas]
I think it would be very powerful to add a playground to the docs where you paste in your own introspection URL, and then it generates the types for you. Maybe it could spin up a monaco editor with types injected.

[Monaco lets you insert global types](https://microsoft.github.io/monaco-editor/playground.html?source=v0.47.0#example-extending-language-services-configure-javascript-defaults) using `addExtraLib`, so you could globally type a `graphql` function based on the generated `.d.ts` file. This would help a lot with adoption and let you get the "aha" moment without needing to spin up a repo, yarn install, etc etc.

**Comment by kitten** [comment]
That's definitely on the list of things I'd like to do.
However, I'm currently waiting for Twoslash to be supported first to get that out of the way ahead of time: https://shikiji.netlify.app/packages/twoslash (Tracked here: https://github.com/expressive-code/expressive-code/issues/95)

Generally, Shiki + Twoslash code samples would cover 90% of our documentation needs. And in other cases I'd love to prefer Stackblitz samples, provided we can get the plugin to work in there, which seems to break in their new version

**Reply by nandorojo** [reply]
Yeah use I Twoslash for [Solito](https://solito.dev), it’s nice. That said, I think there is a lot of power in letting people immediately see their own project in the browser.

I think letting someone paste in their own  introspection doc would be really great. Could also throw it on the homepage and make a cool video out of it.
